tests/network/vmi_infosource.go

```diff
@@ -0,0 +1,198 @@
+
+			defaultBridgeInterface := libvmi.InterfaceDeviceWithBridgeBinding(primaryNetwork)
+			secondaryLinuxBridgeInterface1 := libvmi.InterfaceDeviceWithBridgeBinding(secondaryNetwork1.Name)
+			secondaryLinuxBridgeInterface2 := libvmi.InterfaceDeviceWithBridgeBinding(secondaryNetwork2.Name)
+			vmiSpec := libvmi.NewFedora(
+				libvmi.WithInterface(*libvmi.InterfaceWithMac(&defaultBridgeInterface, primaryInterfaceMac)),
+				libvmi.WithNetwork(kvirtv1.DefaultPodNetwork()),
+				libvmi.WithInterface(*libvmi.InterfaceWithMac(&secondaryLinuxBridgeInterface1, secondaryInterface1Mac)),
+				libvmi.WithInterface(*libvmi.InterfaceWithMac(&secondaryLinuxBridgeInterface2, secondaryInterface2Mac)),
+				libvmi.WithNetwork(secondaryNetwork1),
+				libvmi.WithNetwork(secondaryNetwork2),
+				libvmi.WithCloudInitNoCloudUserData(manipulateGuestLinksScript(primaryInterfaceNewMac, dummyInterfaceMac), false))
+
+			var err error
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmiSpec)
+			Expect(err).NotTo(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(vmi)
+			tests.WaitAgentConnected(virtClient, vmi)
+		})
+
+		It("should have the expected entries in vmi status", func() {
+			expectedInterfaces := []kvirtv1.VirtualMachineInstanceNetworkInterface{
+				{
+					InfoSource: netvmispec.InfoSourceDomain,
+					MAC:        primaryInterfaceMac,
+					Name:       primaryNetwork,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceDomainAndGA,
+					InterfaceName: "eth1",
+					MAC:           secondaryInterface1Mac,
+					Name:          secondaryInterface1Name,
+				},
+				{
+					InfoSource: netvmispec.InfoSourceDomain,
+					MAC:        secondaryInterface2Mac,
+					Name:       secondaryInterface2Name,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceGuestAgent,
+					InterfaceName: "eth0",
+					MAC:           primaryInterfaceNewMac,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceGuestAgent,
+					InterfaceName: dummyInterfaceName,
+					MAC:           dummyInterfaceMac,
+				},
+			}
+
+			var vmiObj *kvirtv1.VirtualMachineInstance
+			Eventually(func() bool {
+				var err error
+				vmiObj, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				return dummyInterfaceExists(vmiObj)
+			}, 120*time.Second, 2*time.Second).Should(Equal(true))
+
+			Expect(len(vmiObj.Status.Interfaces)).To(Equal(len(expectedInterfaces)),
+				fmt.Sprintf("interfaces doesn't match\n%v\n%v", expectedInterfaces, vmiObj.Status.Interfaces))
+
+			orderedInterfaces := orderInterfaceStatus(vmiObj.Status.Interfaces)
+			for i := range orderedInterfaces {
+				expectedInterfaces[i].IP = orderedInterfaces[i].IP
+				expectedInterfaces[i].IPs = orderedInterfaces[i].IPs
+			}
+
+			Expect(reflect.DeepEqual(expectedInterfaces, orderedInterfaces)).To(BeTrue(),
+				fmt.Sprintf("interfaces doesn't match\n%v\n%v", expectedInterfaces, orderedInterfaces))
```

Comment:
same, you need "do".

Regarding the assertion, I think there is a dedicated gomega one you can use instead of this DeepEqual. Then you will not need the message. In fact, `Equal` should be good enough per its description (please confirm it does work).

Reply:
used ConsistOf as the next comment